File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

//品牌接口
public interface BrandService {
    List<TbBrand> findAll();
    PageResult findPage(int pageNum,int pageSize);
    PageResult findPage(TbBrand brand,int pageNum,int pageSize);
    void add(TbBrand brand);
    void update(TbBrand brand);
    //根据id查询实体
    TbBrand findOne(Long id);
    void delete(Long[] ids);
    List<Map> selectOptionList();
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
//分层开发,优点:避免重复代码,易于维护,可读性高
//自定义品牌服务层用于与后台交互,第一个参数是服务名,调用内置服务$http
app.service('brandService',function ($http) {   //品优购服务层
    //示例
    this.search = function (page,rows,searchEntity) {
        return $http.post("../brand/search.do?page="+page+"&rows="+rows,searchEntity);
    }
    this.findOne = function (id) {
        return $http.get("../brand/findOne.do?id="+id);
    }
    this.delete = function (ids) {
        return $http.get('../brand/delete.do?ids='+ids);
    }
    this.save = function (methodName,entity) {
        return $http.post("../brand/"+methodName+".do",entity);
    }
    this.selectOptionList = function () {
        return $http.get('../brand/selectOptionList.do');
    }
});